### ISSUE TITLE
Glide permissions issue with custom system user

### ISSUE THREAD

**Issue by stephenmeehanuk** [statamic/cms#5662]
### Bug description

I'm using Ploi to host a staging site, during set up I decided to create a new system user, and not use the default Ploi user.

This causes problems.

I can upload images and delete them, but they won't show on the website. Looking into the issue it turns out the new system user does't have the same permissions as a default user.

I'm using glide presets, when I upload an image the permissions on the directory is `700`
```
drwx------  2 preview-k7edp preview-k7edp 4096 Mar 26 08:25 test-view06-lr-web-5.jpg
```
I can get the images to show on the website only by changing the permissions via SSH
```
chmod -R 755 test-view06-lr-web-5.jpg
```

### How to reproduce

- Create a test website on Ploi
- Set up a new system user, don't use the default ploi user
- Use glide presets
- Login to website and upload images
- Try to view them on the website

### Logs

```shell
NGINX logs show Permission denied
```

### Versions

Statamic 3.3.2 Pro
Laravel 9.5.1
PHP 8.0.14

### Installation

Other (please explain)
I upgraded from `3.2.38` to `3.3.2`

### Antlers Parser

regex (default)

**Comment by jasonvarga** [comment]
This sounds like a Ploi issue. Statamic doesn't have control over the user. Can you open an issue on their support system?

**Comment by aozisik** [comment]
Just lost ~3 hours getting to the bottom of this. Followed the same journey as @stephenmeehanuk, but I've taken a deep dive in to the vendor directory as well. So here's what I can say:

This isn't a Ploi issue. I can reproduce the same issue on a server provisioned by Laravel Forge. (Ubuntu 20, PHP 8, etc)... In fact, it's possible to observe this even during local development on Mac OS.

By default, the `img` folder for the cached files has permission level `700`. And it's `0600` for the cached images.

<img width="493" alt="Screenshot 2022-03-29 at 22 31 09" src="https://user-images.githubusercontent.com/735011/160692516-4fd72e27-6319-496f-9e11-c251f66a1b25.png">
 
That works without issues with Valet. The problem starts when it's deployed to the server. Nginx refuses to serve images if the files or the containing folders don't have the correct permissions.

So where do these values `700` and `600` defaults come from?  Short answer is in the link below... Private mode is the default mode for flysystem's `LocalFilesystemAdapter`:

https://github.com/thephpleague/flysystem/blob/fd92eb60f71ec4c56ab22406d45a0d0c9c38a8f4/src/UnixVisibility/PortableVisibilityConverter.php#L37-L42


Long answer is:

1. Statamic uses Glide's `ServerFactory` class. Cache path provided here is used to initialise a `LocalFilesystemAdapter` instance.
2. Statamic currently provides this path as a string by default.
https://github.com/statamic/cms/blob/52e19eef345842edba9481c9cfd1876564331dc5/src/Imaging/GlideServer.php#L34-L38
4. When provided as a string, Glide library [initialises the default `LocalFileSystemAdapter` without any extra configuration](https://github.com/thephpleague/glide/blob/bff5b0fe2fd26b2fde2d6958715fde313887d79d/src/ServerFactory.php#L119-L123 ). And let's remember the default is private mode.

 ---

So in conclusion, I think it's Statamic's responsibility to pass a properly configured file system adapter, which sets the visibility mode to "public" rather than private. Currently, I can fix this by returning an adapter instead of a path in `src/Imaging/GlideServer.php`

```php
    public function cachePath()
    {
        return Storage::disk('glide')->getDriver();
    }
```

And my disk config looks like:

```php
        'glide' => [
            'driver' => 'local',
            'root' => public_path('img'),
            'url' => env('APP_URL') . '/img',
            'visibility' => 'public',
        ],
```

@jasonvarga

**Comment by jasonvarga** [comment]
Awesome info. Thanks, we'll look into it.

**Comment by jasonvarga** [comment]
I still find it strange that this issue has only just come up now.

What version of flysystem do you have installed? You can run `composer show | grep flysystem`

In our recent adventure into supporting Laravel 9 and Flysystem 3, we discovered that the default visibility changed - I bet that's the real problem and the reason that you're only running into this now.

**Comment by aozisik** [comment]
It gives back:

```bash
league/flysystem                       3.0.12     File storage abstraction for PHP
```
I've noticed this issue a week ago, which does coincide with us upgrading our build to L9 + Statamic v3.3

Also, here's my final `GlideServer` class, which works without needing a custom disk added to `config/filesystems.php`. Happy to create a PR sometime tomorrow, if you find this workable.

```php
class GlideServer
{
    public function create()
    {
        return ServerFactory::create([
            'source' => base_path(), // this gets overriden on the fly by the image generator
            'cache' => $this->cacheFilesystem(),
            'response' => new LaravelResponseFactory(app('request')),
            'driver' => Config::get('statamic.assets.image_manipulation.driver'),
            'cache_with_file_extensions' => true,
            'presets' => Image::manipulationPresets(),
        ]);
    }

    public function cacheFilesystem()
    {
        $visibility = PortableVisibilityConverter::fromArray([
            'file' => 'public',
            'dir' => 'public',
        ], Visibility::PUBLIC);

        return new Filesystem(
            new LocalFilesystemAdapter(
                $this->cachePath(),
                $visibility
            )
        );
    }

    public function cachePath()
    {
        return Config::get('statamic.assets.image_manipulation.cache')
            ? Config::get('statamic.assets.image_manipulation.cache_path')
            : storage_path('statamic/glide');
    }
}
```

**Comment by aozisik** [comment]
@jasonvarga could you try by using forge’s site isolation feature? that is, by creating the site under a different user other than forge.

**Comment by jasonvarga** [comment]
Ah I saw that checkbox but didn't think that was relevant. Thanks I'll give it a shot.

**Comment by jasonvarga** [comment]
Confirmed! Thanks for all the details. Preparing a fix now.
